### PR TITLE
Issue 93: Add soc_key_container version 5

### PR DESCRIPTION
**Pull request by ericeberry**

### CODE REVIEW

File: reference/src/taimpl/src/porting/otp.c [hunk 1]
```diff
@@ -265,8 +266,19 @@ static sa_status otp_hw_key_ladder(
...
-        if (!get_root_key(root_key, &root_key_length)) {
-            ERROR("get_root_key failed");
+        if (root_key_type == UNIQUE) {
+            if (!get_root_key(root_key, &root_key_length)) {
+                ERROR("get_root_key failed");
+                break;
+            }
+        } else if (root_key_type == COMMON) {
+            if (!get_common_root_key(root_key, &root_key_length)) {
+                ERROR("get_root_key failed");
```
Comment (mhabrat) [inline reference/src/taimpl/src/porting/otp.c:276]: get_common_root_key?

File: reference/src/taimpl/src/ta_sa_key_derive.c [hunk 2]
```diff
@@ -106,14 +106,14 @@ static sa_status ta_sa_key_derive_root_key_ladder(
...
             if (status != SA_STATUS_OK) {
                 ERROR("otp_root_key_ladder failed");
                 break;
             }
         } else if (kdf_algorithm == SA_KDF_ALGORITHM_COMMON_ROOT_KEY_LADDER) {
-            status = otp_common_root_key_ladder(&stored_key_derived, rights, parameters->c1, parameters->c2,
+            status = otp_root_key_ladder(&stored_key_derived, rights, COMMON, parameters->c1, parameters->c2,
                     parameters->c3, parameters->c4);
             if (status != SA_STATUS_OK) {
                 ERROR("otp_common_root_key_ladder failed");
```
Comment (mhabrat) [inline reference/src/taimpl/src/ta_sa_key_derive.c:119]: otp_root_key_ladder?

**Review by mhabrat — CHANGES_REQUESTED** [review]

File: reference/src/taimpl/src/porting/otp.c (hunk 1 above)
Comment (ericeberry) [reply reference/src/taimpl/src/porting/otp.c:276]: Done

File: reference/src/taimpl/src/ta_sa_key_derive.c (hunk 2 above)
Comment (ericeberry) [reply reference/src/taimpl/src/ta_sa_key_derive.c:119]: Done

**Review by mhabrat — APPROVED** [review]

**Pull request merged**